src/util.ts

```diff
@@ -1,45 +1,62 @@
 import * as core from '@actions/core';
 import * as exec from '@actions/exec';
 
-export function parseNodeVersionFile(contents: string): string | null {
-  let nodeVersion: string | undefined;
+import fs from 'fs';
+import path from 'path';
+
+export function getNodeVersionFromFile(versionFilePath: string): string | null {
+  if (!fs.existsSync(versionFilePath)) {
+    throw new Error(
+      `The specified node version file at: ${versionFilePath} does not exist`
+    );
+  }
+
+  const contents = fs.readFileSync(versionFilePath, 'utf8');
 
   // Try parsing the file as an NPM `package.json` file.
   try {
     const manifest = JSON.parse(contents);
 
-    // JSON can parse numbers, but that's handled later
-    if (typeof manifest === 'object') {
-      nodeVersion = manifest.volta?.node;
-      if (!nodeVersion) nodeVersion = manifest.engines?.node;
+    // Presume package.json file.
+    if (typeof manifest === 'object' && manifest !== null) {
```

Comment:
Just a nitpick. Could you please change `manifest !== null` to `!!manifest`

Reply:
Done!